src/datanode/src/sql/delete.rs

```diff
@@ -0,0 +1,139 @@
+            table_name: table_ref.to_string(),
+        })?;
+
+        Ok(Output::AffectedRows(affected_rows))
+    }
+
+    pub(crate) fn delete_to_request(
+        &self,
+        stmt: Delete,
+        query_ctx: QueryContextRef,
+    ) -> Result<SqlRequest> {
+        let (catalog_name, schema_name, table_name) =
+            table_idents_to_full_name(stmt.table_name(), query_ctx)?;
+        let key_column_values = parser_selection(stmt.selection())?;
+        Ok(SqlRequest::Delete(DeleteRequest {
+            key_column_values,
+            catalog_name,
+            schema_name,
+            table_name,
+        }))
+    }
+}
+
+/// parse selection, currently supported format is `tagkey1 = 'tagvalue1' and tagkey2 = 'tagvalue2'`.
+/// (only uses =, and in the where clause and provides all columns needed by the key.)
+fn parser_selection(selection: &Option<Expr>) -> Result<HashMap<String, VectorRef>> {
+    let mut key_column_values = HashMap::new();
+    if let Some(expr) = selection {
+        parser_expr(expr, &mut key_column_values)?;
+    }
+    Ok(key_column_values)
+}
+
+fn parser_expr(expr: &Expr, key_column_values: &mut HashMap<String, VectorRef>) -> Result<()> {
+    // match BinaryOp
+    if let Expr::BinaryOp { left, op, right } = expr {
+        match (left.deref(), op, right.deref()) {
+            // match And operator
+            (Expr::BinaryOp { .. }, BinaryOperator::And, Expr::BinaryOp { .. }) => {
+                parser_expr(left.deref(), key_column_values)?;
+                parser_expr(right.deref(), key_column_values)?;
+                return Ok(());
+            }
+            // match Eq operator
+            (Expr::Identifier(column_name), BinaryOperator::Eq, Expr::Value(value)) => {
+                key_column_values.insert(
+                    column_name.to_string(),
+                    value_to_vector(&column_name.to_string(), value)?,
+                );
+                return Ok(());
+            }
+            (Expr::Identifier(column_name), BinaryOperator::Eq, Expr::Identifier(value)) => {
+                key_column_values.insert(
+                    column_name.to_string(),
+                    Arc::new(StringVector::from(vec![value.to_string()])),
+                );
+                return Ok(());
+            }
+            _ => {}
+        }
+    }
+    return InvalidSqlSnafu {
+        msg: format!("Failed to parse expr:{expr}"),
+    }
+    .fail();
+}
+
+/// parse value to vector
+fn value_to_vector(column_name: &String, value: &Value) -> Result<VectorRef> {
+    match value {
+        Value::Number(n, _) => {
+            if column_name == "ts" {
+                Ok(Arc::new(TimestampMillisecondVector::from_vec(vec![
+                    i64::from_str(n).unwrap(),
+                ])))
+            } else {
+                Ok(Arc::new(Float64Vector::from_vec(vec![
+                    f64::from_str(n).unwrap()
+                ])))
+            }
```

Comment:
The assumed column type might be incorrect. Timestamp columns could use arbitrary names, and use data types like Int64. You should find the data type from `Schema` and use [create_mutable_vector](https://github.com/GreptimeTeam/greptimedb/blob/c0d3533d10037d39d54f14cf22d6bb1cffdccbe5/src/datatypes/src/data_type.rs#L264) to create vectors with correct type.

The [`sql_value_to_value `](https://github.com/GreptimeTeam/greptimedb/blob/c0d3533d10037d39d54f14cf22d6bb1cffdccbe5/src/sql/src/statements.rs#L204) function might be useful to convert a SQL value into our value.

Reply:
Ask a question unrelated to pr. Call sql's function `sql_value_to_value ` in `datanode` module, because the two modules use their own `Result` and `Error`, cannot use the `?` directly, and needs to be converted again. 
Is there any good method or unified to the `common` module?

Reply:
We can hold the source error in the enum variant. There is already a variant for this case so you could just use [context](https://docs.rs/snafu/latest/snafu/trait.ResultExt.html#tymethod.context) to convert the error.
https://github.com/GreptimeTeam/greptimedb/blob/e77a7f253c1473b19ad2bda48305299daa7524f2/src/datanode/src/error.rs#L104-L108

[`datanode::Error`](https://github.com/GreptimeTeam/greptimedb/blob/37ca5ba380b6f22df078fbd86da3062548ae1cdb/src/datanode/src/error.rs) itself should be a good example.